### ISSUE TITLE
Indent rule problem with object and array literals

### ISSUE THREAD

**Issue by targos** [eslint/eslint#3236]
Version 1.0.0

Rule:
```yml
indent: [2, 2, {"SwitchCase": 1, "VariableDeclarator": 2}]
```

Code:
```js
var testCases =
    [{ crl: 'ca2-crl',
      clients:
       [
        { name: 'agent1', shouldReject: true, shouldAuth: false },
        { name: 'agent2', shouldReject: true, shouldAuth: false }
       ]
    }
    ];

var abc =
     {
       x: 2,
        y: 4
     };
```

Result:
```
test1.js
   5:9  error  Expected indentation of 9 characters but found 8  indent
   6:9  error  Expected indentation of 9 characters but found 8  indent
  14:9  error  Expected indentation of 7 characters but found 8  indent
```

The rule accepts any indentation for the object or array literal if it starts on a new line. Then it uses this indentation as a reference for the next errors, so we see an expected odd indentation (9, 7) when the rule is set to 2.

Cross-reference: https://github.com/nodejs/io.js/pull/2286

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!
